core/src/main/java/org/elasticsearch/plugins/InstallPluginCommand.java

```diff
@@ -134,6 +134,26 @@
     private final OptionSpec<Void> batchOption;
     private final OptionSpec<String> arguments;
 
+    public static final Set<PosixFilePermission> dirAndExecutablePerms = new HashSet<>(7);
+    public static final Set<PosixFilePermission> filePerms = new HashSet<>(4);
+
+    static {
+        // Directories and executables get chmod 755
+        dirAndExecutablePerms.add(PosixFilePermission.OWNER_EXECUTE);
```

Comment:
Can we create a local set, add the permissions to that set, and then make `dirAndExecutablePerms` an unmodifiable copy of the local set?